src/leap/mail/adaptors/soledad.py

```diff
@@ -483,17 +529,24 @@ def create(self, store):
 
         # TODO check that the doc_ids in the mdoc are coherent
         d = []
-        d.append(self.mdoc.create(store))
+        mdoc_created = self.mdoc.create(store)
+        d.append(mdoc_created)
         d.append(self.fdoc.create(store))
-        if self.hdoc.doc_id is None:
-            d.append(self.hdoc.create(store))
-        for cdoc in self.cdocs.values():
-            if cdoc.doc_id is not None:
-                # we could be just linking to an existing
-                # content-doc.
-                continue
-            d.append(cdoc.create(store))
-        return defer.gatherResults(d)
+
+        if not self._is_copy:
+            if self.hdoc.doc_id is None:
+                d.append(self.hdoc.create(store))
+            for cdoc in self.cdocs.values():
+                if cdoc.doc_id is not None:
+                    # we could be just linking to an existing
+                    # content-doc.
+                    continue
+                d.append(cdoc.create(store))
+
+        if notify_just_mdoc:
+            return mdoc_created
+        else:
+            return defer.gatherResults(d)
```

Comment:
AFAIK if you don't add errbacks and the deferred fails the program will crash when the garbage collector tries to free it. For the case of notify_just_mdoc we might need to add a dummy errback.